design-proposals/sidecar-hooks.md

```diff
@@ -0,0 +1,56 @@
+# Sidecar hooks
+
+## Overview
+
+A hooking mechanism that allows customizations to the VMI before it starts without changing KubeVirt
+core components. The common user of those hooks are Sidecars that will run in the same Pod that the
+VMI will be deployed.
+
+## Motivation
+
+A way to accommodate use cases that we might not want to bring and support in KubeVirt itself or to
+give room to test and develop features while they are still not mature enough.
+
+## Goals
+
+- Provide and maintain a set of gRPC APIs for users to do modifications to the VMI
+- Provide and maintain sidecars that are important for core functionalities
+
+## Non Goals
+
+- Provide and maintain a wide variety of sidecars that are not part of KubeVirt's core
+  functionalities.
+
+## User Stories
+
+- As a VM owner, I want to apply custom configurations to QEMU command line
```

Comment:
It isn't really only for QEMU commandline. This is a single option. More in general is for providing custom XML portions